cdap-proto/src/main/java/co/cask/cdap/proto/artifact/ArtifactRange.java

```diff
@@ -96,7 +96,11 @@ public int hashCode() {
 
   @Override
   public String toString() {
-    return new StringBuilder()
+    StringBuilder stringBuilder = new StringBuilder();
+    if (namespace.getId().equals(Id.Namespace.SYSTEM)) {
```

Comment:
i think it was because the handler only deals with ranges within its own namespace and with system namespace.  But that logic shouldn't be a part of this class, will add namespace no matter what.